test/test_deployment.py

```diff
@@ -0,0 +1,682 @@
+    """Start mock GitHub API server
+
+    Return environment shell command for using it
+    """
+    subprocess.run(['podman', 'cp', str(ROOT_DIR / 'tasks/mock-github'), f'{pod.tasks}:/work/bots/mock-github'],
+                   check=True)
+    mock_pid = exec_c_out(pod.tasks, f'''
+        cd bots
+        PYTHONPATH=. nohup ./mock-github --log /tmp/mock.log cockpit-project/bots {bots_sha} \
+                </dev/null > /tmp/mock.out 2>&1 &
+        echo $!''').strip()
+    # wait until it started
+    exec_c_out(pod.tasks, f'curl --retry 5 --retry-connrefused --fail {GHAPI_URL_POD}/repos/cockpit-project/bots')
+
+    yield f'export GITHUB_API={GHAPI_URL_POD}; SHA={bots_sha}'
+
+    exec_c_out(pod.tasks, f'while kill {mock_pid}; do sleep 0.1; done')
+
+
+@pytest.fixture()
+def mock_git_push(pod: PodData) -> Generator[None, None, None]:
+    """Install tasks/mock-git-push"""
+
+    subprocess.run(['podman', 'cp', str(ROOT_DIR / 'tasks/mock-git-push'), f'{pod.tasks}:/usr/local/bin/git'],
+                   check=True)
+
+    yield None
+
+    exec_c(pod.tasks, 'rm /usr/local/bin/git', exec_opts=['--user', 'root'])
+
+
+def generate_config(config: Config, forge_opts: str, run_args: str) -> None:
+    conf = textwrap.dedent(f'''\
+        [logs]
+        driver='s3'
+
+        [forge.github]
+        token = [{{file="/run/secrets/webhook/.config--github-token"}}]
+        {forge_opts}
+
+        [logs.s3]
+        url = '{S3_URL_POD}/logs'
+        ca = [{{file='/run/secrets/webhook/ca.pem'}}]
+        key = [{{file="/run/secrets/tasks/s3-keys/localhost.localdomain"}}]
+
+        [container]
+        command = ['podman-remote', '--url=unix:///podman.sock']
+        run-args = [
+            '--security-opt=label=disable',
+            '--volume={ROOT_DIR / 'tasks'}/mock-git-push:/usr/local/bin/git:ro',
+            '--env=COCKPIT_IMAGE_UPLOAD_STORE={S3_URL_POD}/images/',
+            '--env=GIT_AUTHOR_*',
+            '--env=GIT_COMMITTER_*',
+            {run_args}
+        ]
+
+        [container.secrets]
+        # these are *host* paths, this is podman-remote
+        image-upload=[
+            '--volume={config.tasks}/s3-keys:/run/secrets/s3-keys:ro',
+            '--env=COCKPIT_S3_KEY_DIR=/run/secrets/s3-keys',
+            '--volume={config.webhook}/ca.pem:/run/secrets/ca.pem:ro',
+            '--env=COCKPIT_CA_PEM=/run/secrets/ca.pem',
+        ]
+        github-token=[
+            '--volume={config.webhook}/.config--github-token:/run/secrets/github-token:ro',
+            '--env=COCKPIT_GITHUB_TOKEN_FILE=/run/secrets/github-token',
+        ]
+        ''')
+
+    (config.tasks / 'job-runner.toml').write_text(conf)
+
+
+@pytest.fixture()
+def mock_runner_config(config: Config, pod: PodData) -> None:
+    generate_config(config,
+                    forge_opts=f'api-url = "{GHAPI_URL_POD}"',
+                    run_args=f'"--pod={pod.pod}", "--env=GITHUB_API={GHAPI_URL_POD}"')
+    return None  # silence ruff PT004
```

Comment:
I'd much rather these returned a path object which we could pass to `job-runner -F`...

Reply:
But then we'd diverge from what happens in the actual deployment, where we don't use `-F`, and instead point to it via env variable. I can return the path of course, but it's not actually going to use it. It looks a bit nicer, though.